test/functions/iterators/diten/iteratorNoYield.good

```diff
@@ -1 +1 @@
-iteratorNoYield.chpl:9: error: illegal use of function that does not return a value: 'itr'
+iteratorNoYield.chpl:9: error: cannot iterate over values of type void
```

Comment:
I was thinking about this case a bit more this morning, and it increasingly seems to me that this might be the compelling case for distinguishing between "actually yields voids" vs. "contains no yields."  Consider:

````chapel
iter yieldTenVoids() {
  for i in 1..10 do
    yield _void;
}

iter yieldAVoid() {
  yield _void;
}

iter yieldNothing() {
  writeln("In my iterator!")
}
````

I strongly suspect that it would be surprising if the following loops did not behave as indicated in the comments:

````chapel
for v in yieldTenVoids() do
  writeln("Got a void!");    // should print this message ten times

for v in yieldAVoid() do
  writeln("Got a void!");    // should print this message one time

for v in yieldNothing() do
  writeln("Got a void!");    // should not print the loop body's message at all, but should still print the iterator's message
````

Does it seem correct that that would be the right behavior?  Like the "return nothing" error message change comments above, it seems plausible that we could implement this by having the compiler attach a flag to the iterator to distinguish between "void because it returns/yields a void" vs. "void because it returns/yields nothing."

As with my other comments, I wouldn't hold the PR up over this, but at present, this seems like the logical way to take these cases in the future.